### PR TITLE
Modify search endpoint to GET request

### PR DESCRIPTION
**Pull request by patriciaahuang**
## Overview

Changed the search rides endpoint from a POST request to a GET request, since no data is modified.

## Changes Made

I changed the endpoint from "/search" to:

"/search/depart/<str:depart>/start/<str:start>/end/<str:end>/radius/<int:radius>/" 

which includes all the previous request body arguments. I added a method to be able to get data from multiple URL parameters and not just one (by default). After getting the data from the URL parameters, the same controller is used to process the data. I also modified the User serializer to not delete rides that are past the current date to preserve data.

## Test Coverage

Using Postman, I created a new ride at "/rides" then created a GET request for "/search/depart/<str:depart>/start/<str:start>/end/<str:end>/radius/<int:radius>/" with parameters that matched the ride (which returned the created ride) and also parameters that did not match the ride (which returned no rides).

### CODE REVIEW

File: src/person/serializers.py [hunk 1]
```diff
@@ -62,8 +62,6 @@ def get_rides(self, user):
         for ride in rides:
             if ride.departure_datetime >= timezone.now():
                 active_rides.add(ride)
-            else:
-                ride.delete()
```
Comment (joycrwu) [inline src/person/serializers.py]: is this different than what noah had deleted in his PR?

Comment (patriciaahuang) [reply src/person/serializers.py]: it should be the same

**Review by joycrwu — APPROVED** [review]

**Comment by maxwell1231** [comment]
Can we change it to time instead of depart

**Review by maxwell1231 — APPROVED** [review]
LGTM!

**Pull request merged**